universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java

```diff
@@ -132,6 +147,41 @@ public void putString_writesNullIfStringIsEmpty() {
     assertThat(data.readString(path)).isEmpty();
   }
 
+  @Test
+  public void putString_withFirstRepeatedEntity_putsParentLists() {
+    ApplicantData data = new ApplicantData();
+
+    data.putString(Path.create("applicant.children[0].favorite_color.text"), "Orange");
+
+    String expected =
+        "{\"applicant\":{\"children\":[{\"favorite_color\":{\"text\":\"Orange\"}}]},\"metadata\":{}}";
+    assertThat(data.asJsonString()).isEqualTo(expected);
+  }
+
+  @Test
+  public void putString_withSecondRepeatedEntity_addsIt() {
+    ApplicantData data = new ApplicantData();
+
+    data.putString(Path.create("applicant.children[0].favorite_color.text"), "Orange");
+    data.putString(Path.create("applicant.children[1].favorite_color.text"), "Brown");
+
+    String expected =
+        "{\"applicant\":{\"children\":[{\"favorite_color\":{\"text\":\"Orange\"}},{\"favorite_color\":{\"text\":\"Brown\"}}]},\"metadata\":{}}";
+    assertThat(data.asJsonString()).isEqualTo(expected);
+  }
+
+  // TODO(#624): get rid of this recursion
```

Comment:
comment: I'm not convinced we can enforce array index ordering on ApplicantData - it may be the job of the caller to assert it is in a good state? I'm not sure

Reply:
I'll add this comment to #624, and let whoever works on that figure it out.